lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -159,14 +352,41 @@ def filter_query(
         'saved': True}` because there's no mechanism to parse JSON expressions.
         This might be added in the future.
 
-        To match against a key value, separate the key path from the value with
-        a `:` character, e.g., "dataset.name:fio". This produces an exact match
-        expression as an AND (required) term in the query: only datasets with
-        the exact name "fio" will be returned.
-
-        To match against a subset of a key value, insert a tilde (~) following
-        the `:`, like "dataset.name:~fio". This produces a "contains" filter
-        that will match against any name containing the substring "fio".
+        Specify the value to match separated from the key path by a `:`
+        character, e.g., "dataset.name:fio". By default this produces an exact
+        match expression as an AND (required) term in the query: only datasets
+        with the exact name "fio" will be returned.
+
+        The value may be prefixed by an optional operator in order to perform a
+        comparison other than strict equality:
+
+        =   Look for matches strictly equal to the specified value (default)
+        ~   Look for matches containing the specified value as a substring
+        >   Look for matches strictly greater than the specified value
+        <   Look for matches strictly less than the specified value
+        >=  Look for matches greater than or equal to the specified value
+        <=  Look for matches less than or equal to the specified value
+        !=  Look for matches not strictly equal to the specified value
+
+        After the value, you can optionally specify a type for the comparison.
+        Note that an incompatible type (other than the default "str") for a
+        primary "dataset" column will be rejected, but general Metadata
+        (including "dataset.metalog") is stored as generic JSON and the API
+        will attempt to cast the selected data to the specified type.
+
+        str     Perform a string match
+        int     Perform an integer match
+        bool    Perform a boolean match (boolean values are t[rue], f[alse],
+                y[es], and n[o])
+        date    Perform a date match: the selected key must have a string value
+                in ISO-8601 format, and will be interpreted as UTC if no time
+                zone is specified.
```

Comment:
Doesn't ISO-8601 sort lexicographically, such that conversion from string to date doesn't accomplish anything?  (I suppose it would allow for comparing values with different timezone offsets....)

Reply:
Looking at the tests, apparently the input string can have other formats (like "US Standard"...), so maybe this text needs updating?